yarn-project/prover-node/src/prover-node.ts

```diff
@@ -216,7 +216,7 @@ export class ProverNode implements EpochMonitorHandler, ProverNodeApi, Traceable
 
     // Fast forward world state to right before the target block and get a fork
     this.log.verbose(`Creating proving job for epoch ${epochNumber} for block range ${fromBlock} to ${toBlock}`);
-    await this.worldState.syncImmediate(fromBlock - 1);
+    await this.worldState.syncImmediate(toBlock);
```

Comment:
This is so we can check that the resulting state of re-executing a block for proving yields the expected world state roots? Otherwise some blocks may still not be on world-state?